packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -214,22 +219,24 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     });
   }
 
-  protected async loadFunds(infoHash: string, peerId: string, channelId: string) {
-    // [ George ] If web3torrent is to run the Single Asset Payments ForceMoveApp, and the payments are going to be unidirectional we could wrap the paymentChannelClient in a web3tChannelClient which offers a countersign() convenience method. This allows the seeder to immediately accept the payment and for the leecher to be ready to send another one as quickly as possible.
-
-    log(`querying channel client for updated buffer`);
+  protected async refillBuffer(infoHash: string, peerId: string, channelId: string) {
+    log(`querying channel client for updated balance`);
     const newSeederBalance = bigNumberify(
       this.paymentChannelClient.channelCache[channelId].beneficiaryBalance
     );
+    // infer payment using update balance and previously stored balance
     const payment = newSeederBalance.sub(
       bigNumberify(this.peersList[infoHash][peerId].seederBalance)
     );
+    // store new balance
+    this.peersList[infoHash][peerId].seederBalance = newSeederBalance.toString();
+    // convert payment into number of pieces
     this.peersList[infoHash][peerId].buffer = bigNumberify(this.peersList[infoHash][peerId].buffer)
-      .add(payment)
+      .add(payment.div(WEI_PER_PIECE)) // TODO allow this to vary by piece size
```

Comment:
We should ensure `buffer` remains an integer, since our check for choking is now `.isZero` and we decrement by `1`.